Add Guild object (#25)
Arcader717/Async-DisOAuth2: docs/source/api.rst

      .. versionadded:: 1.1
        Used to get the users guilds.

    .. py:method:: get_guild(id, with_counts)
      :async:

      Gets a guild from the Guild ID

      :param id: The id of the guild you want
      :param with_counts: an optional parameter that indicates you want (True) Discord to return :py:attr:`approximate_member_count` :py:attr:`approximate_presence_count` or not (False). Defaults to False
      :type: bool or None
      :type: int or None
      :return: The guild object for the given id
      :rtype: :class:`GuildObj`

>     

Author: ```suggestion
    
    .. versionadded:: 1.2
        Based off of the ``GET /guilds/<id>`` endpoint
```

  .. property:: max_members

    the maximum number of members for the guild

    :type: int or None

  .. property:: vanity_url_code

    the vanity url code for the guild

    :type: str or None

  .. property:: description

    the description of the guild

    :type: str or None

  .. property:: banner

    The banner hash

    :type: str or None

  .. property:: premium_tier

    The guild's premium tier (Server Boost level)

    :type: int

  .. property:: premium_subscription_count

    the number of boosts this guild currently has

    :type: int or None

  .. property:: preferred_locale

    The preferred locale of a community guild; userd in server discovery and notices from Discord, and sent in interactions; defaults to "en-US"

    :type: str

  .. property:: public_updates_channel_id

    The id of the channel where admins and moderators of Community guilds recieve notices from Discord

  .. property:: max_video_channel_users

    The maximum amount of users in a video channel

    :type: int

  .. property:: max_stage_video_channel_users

    the maximum amount of users in a stage video channel

    :type: int

  .. property:: approximate_member_count

>     approximate number of members in this guild, returned from the ``GET /guilds/<id>`` and ``/users/@me/guilds`` endpoints when ``with_counts`` is ``true``

Author: Add a function to use the `GET /guilds/<id>` endpoint, add the parameters to each request as an Optional, and then update the documentation to be about the new functions

>     approximate number of members in this guild, returned from the ``GET /guilds/<id>`` and ``/users/@me/guilds`` endpoints when ``with_counts`` is ``true``

Author: ```suggestion
    approximate number of members in this guild, returned from the ``GET /guilds/<id>`` and ``/users/@me/guilds`` endpoints when ``with_counts`` is ``true``
```

Add a function to use the `GET /guilds/<id>` endpoint, add the parameters to each request as an Optional, and then update the documentation to be about the new functions

    The user's email

    :type: str or None
    
  .. property:: flags

    The flags on the user's account

    :type: int or None
    
  .. property:: premium_type

    The type of nitro the user has

    :type: int or None
    
  .. property:: public_flags

    The public flags on a user's account

    :type: int or None
    
  .. property:: avatar_decoration

    The user's avatar decoration hash

    :type: str or None
    

.. class:: GuildObj

>   .. versionadded:: 1.2.0

Author: ```suggestion
  .. versionadded:: 1.2
```

    .. depreciated:: 
      This was only added to follow the guild object from Discord. Replaced by channel.rtc_region

    the region of the guild

    :type: str or None

  .. property:: afk_channel_id

    the channel id where afk users go

  .. property:: afk_timeout

    The time it takes for a afk user to be sent to the afk channel

  .. property:: widget_enabled

    Whether a widget is enabled or not

    :type: bool or None

  .. property:: widget_channel_id

    The channel the widget is in

  .. property:: verification_level

    The level of verification in the guild

    :type: int or None

  .. property:: default_message_notifications

    The default notification level of the guild

    :type: int or None

  .. property:: explicit_content_filter

    The content filter level of the guild

    :type: int or None

  .. property:: roles

    The roles of the guild, as an array

  .. property:: emojis

    The emojis of the guild, as an array of emoji object

  .. property:: features

    The features of the guild

  .. property:: mfa_level

    The required level of MFA (Multi-Factor Authentication) for the guild

    :type: int

  .. property:: application_id

>     ID of the application that made the guilf, if it is bot-created

Author: ```suggestion
    ID of the application that made the guild, if it is bot-created
```
typo fixes

    the description of the guild

    :type: str or None

  .. property:: banner

    The banner hash

    :type: str or None

  .. property:: premium_tier

    The guild's premium tier (Server Boost level)

    :type: int

  .. property:: premium_subscription_count

    the number of boosts this guild currently has

    :type: int or None

  .. property:: preferred_locale

    The preferred locale of a community guild; userd in server discovery and notices from Discord, and sent in interactions; defaults to "en-US"

    :type: str

  .. property:: public_updates_channel_id

    The id of the channel where admins and moderators of Community guilds recieve notices from Discord

  .. property:: max_video_channel_users

    The maximum amount of users in a video channel

    :type: int

  .. property:: max_stage_video_channel_users

    the maximum amount of users in a stage video channel

    :type: int

  .. property:: approximate_member_count

    approximate number of members in this guild, returned from the ``GET /guilds/<id>`` and ``/users/@me/guilds`` endpoints when ``with_counts`` is ``true``

    :type: int

  .. property:: approximate_presence_count

>     approximate number of non-offline members in this guild, returned from the ``GET /guilds/<id>`` and ``/users/@me/guilds`` endpoints when ``with_counts`` is ``true``

Author: ```suggestion
    approximate number of non-offline members in this guild, returned from the :meth:`get_guild` and :meth:`get_user_guilds` when ``with_counts`` is ``True``
```
